iocage_lib/ioc_plugin.py

```diff
@@ -469,11 +486,31 @@ def __fetch_plugin_props__(self, conf, props, num):
         # supplied properties replacing ours.
         create_props = [
             f'release={release}', 'boot=on', 'vnet=1'
+        ] + [
+            f'{k}={v}' for k, v in props.items()
         ]
 
-        create_props = create_props + [
-            f"{k}={v}" for k, v in (p.split("=") for p in props)
-        ]
+        if all(
+            props.get(k, 'none') == 'none'
+            for k in ('ip4_addr', 'ip6_addr')
+        ) and (not iocage_lib.ioc_common.lowercase_set(
+                iocage_lib.ioc_common.construct_truthy('dhcp')
+        ) & iocage_lib.ioc_common.lowercase_set(
+            create_props) and not iocage_lib.ioc_common.lowercase_set(
+            iocage_lib.ioc_common.construct_truthy('ip_hostname')
+        ) & iocage_lib.ioc_common.lowercase_set(
+            create_props) and not iocage_lib.ioc_common.lowercase_set(
+            iocage_lib.ioc_common.construct_truthy('nat')
+        ) & iocage_lib.ioc_common.lowercase_set(create_props)):
+            iocage_lib.ioc_common.logit(
```

Comment:
I regret writing blocks like this, perhaps we should make this a common function so it reads much nicer?

Reply:
Roger that :p